modules/core/jvm/src/main/scala/SSLPlatform.scala

```diff
@@ -11,60 +11,15 @@ import java.security.KeyStore
 import javax.net.ssl.SSLContext
 import fs2.io.net.Network
 import fs2.io.net.tls.TLSContext
-import fs2.io.net.tls.TLSParameters
-import skunk.net.SSLNegotiation
-
-sealed abstract class SSL(
-  val tlsParameters: TLSParameters = TLSParameters.Default,
-  val fallbackOk:    Boolean       = false,
-) { outer =>
-
-  def tlsContext[F[_]: Network](implicit ev: ApplicativeError[F, Throwable]): F[TLSContext[F]]
-
-  def withTLSParameters(tlsParameters: TLSParameters): SSL =
-    new SSL(tlsParameters, fallbackOk) {
-      def tlsContext[F[_]: Network](implicit ev: ApplicativeError[F, Throwable]): F[TLSContext[F]] =
-        outer.tlsContext
-    }
-
-  def withFallback(fallbackOk: Boolean): SSL =
-    new SSL(tlsParameters, fallbackOk) {
-      def tlsContext[F[_]: Network](implicit ev: ApplicativeError[F, Throwable]): F[TLSContext[F]] =
-        outer.tlsContext
-    }
-
-  def toSSLNegotiationOptions[F[_]: Network](logger: Option[String => F[Unit]])(
-    implicit ev: ApplicativeError[F, Throwable]
-  ): F[Option[SSLNegotiation.Options[F]]] =
-    this match {
-      case SSL.None => none.pure[F]
-      case _ => tlsContext.map(SSLNegotiation.Options(_, tlsParameters, fallbackOk, logger).some)
-    }
-
-}
-
-object SSL {
-
-  /** `SSL` which indicates that SSL is not to be used. */
-  object None extends SSL() {
-    def tlsContext[F[_]: Network](implicit ev: ApplicativeError[F, Throwable]): F[TLSContext[F]] =
-      ev.raiseError(new Exception("SSL.None: cannot create a TLSContext."))
-    override def withFallback(fallbackOk: Boolean): SSL = this
-    override def withTLSParameters(tlsParameters: TLSParameters): SSL = this
-  }
 
+private[skunk] trait SSLCompanionPlatform { self: SSL.type =>
```

Comment:
If you're not using the `self` alias somewhere you can use `this: SSL.type =>` instead.